libr/core/cmd_search.c

```diff
@@ -1924,6 +1924,9 @@ static void do_string_search(RCore *core, struct search_parameters *param) {
 		r_cons_printf ("fs hits\n");
 	}
 	core->search->inverse = param->inverse;
+	if (param->inverse) {
+		core->search->maxhits = 1;
```

Comment:
This behavior was implemented in `R_API int r_search_mybinparse_update(void *_s, ut64 from, const ut8 *buf, int len) {` and there are tests testing this. I just moved it outside .